tests/fixture/pageObject/playwright/PollyObject.ts

```diff
@@ -0,0 +1,160 @@
+                }
+                if (checkoutIdString) {
+                    return { checkoutId: checkoutIdString };
+                }
+
+                // Critical error
+                throw new Error('Unable to find checkoutId from the \'/api/storefront/checkout-settings\' request recording.');
+            } else {
+                throw new Error('Unable to find any recording from the located HAR file.');
+            }
+        } else {
+            throw new Error('Unable to locate the HAR file folder.');
+        }
+    }
+
+    start(option: PollyOptions): void {
+        const { mode, page, har, storeUrl, devMode } = option;
+
+        Polly.register(PlaywrightAdapter);
+        Polly.register(FSPersister);
+
+        this.polly = new Polly(har);
+        this.polly.configure({
+            mode: this.mode,
+            logLevel: devMode ? 'INFO' : 'SILENT',
+            recordIfMissing: false,
+            flushRequestsOnStop: true,
+            recordFailedRequests: true,
+            adapters: ['playwright'],
+            adapterOptions: {
+                playwright: {
+                    context: page,
+                },
+            },
+            persister: CustomFSPersister,
+            persisterOptions: {
+                keepUnusedRequests: false,
+                disableSortingHarEntries: true,
+                CustomFSPersister: {
+                    recordingsDir: path.join(__dirname, '../../../har/'),
+                },
+            },
+            matchRequestsBy: {
+                method: true,
+                url: true,
+                order: true,
+                headers: false,
+                body: true,
+            },
+        });
+
+        this.polly.server.any().on('request', req => {
+            // To ensure that requests match HAR entries, ignore sensitive and constantly changing headers/payloads
+            req.headers = Object.fromEntries(
+                Object.entries(req.headers).filter(([key, _]) => !includes(ignoredHeaders, key))
+            );
+
+            if (req.body && req.body.length > 0) {
+                req.body = JSON.stringify(this.sortPayload(req.jsonBody()));
+            }
+        });
+
+        if (mode === 'replay') {
+            this.polly.server.get(this.baseUrl + '/api/storefront/checkout-settings').on('beforeResponse', (_, res) => {
+                // bigpayBaseUrl must be the exact URL of the local environment in order for Bigpay iframes to work.
+                const response = res.jsonBody();
+                response.storeConfig.paymentSettings.bigpayBaseUrl = this.baseUrl;
+                res.send(response);
+            });
+            this.polly.server.any().on('request', req => {
+                // To ensure that requests match HAR entries, change local URLs to production URLs.
+                if (includes(req.url, '/api/public/v1/orders/payments')) {
+                    req.url = req.url.replace(this.baseUrl, 'https://bigpay.service.bcdev');
```

Comment:
The BigPay host is environment dependent as well so we won't be able to hard-code it.

Reply:
I found `/api/public/v1/orders/payments` is hard coded in [bigpay-client-js](https://github.com/bigcommerce/bigpay-client-js/blob/455dc9b98b48542403708359fc9d2fb6320daf19/src/payment/url-helper.js#L42). So I used this pathname to identify bigpay request among HAR recordings.

To better support the replay mode, I added the `getBigpayBaseUrl()` and `enableReplay()` methods to `PollyObject`.

Also, I set the sample tests folder as an ignore folder in `playwright`, so `npx playwright test` won't run them, but `npx playwright test sampletests` will.